samples/Kconfig

```diff
@@ -16,15 +16,54 @@ config NCS_SAMPLES_DEFAULTS
 	help
 	  Use the default configuration for NCS samples.
 
-config NCS_SAMPLE_EMPTY_APP_CORE_CHILD_IMAGE
+if SOC_SERIES_NRF53X
+
+config AUTO_INCLUDE_CHILD_IMAGE
+	bool "Automatically add a child image when relevant"
+	select PARTITION_MANAGER_ENABLED
+	help
+	  Add a child image to the other core. If the current sample is built
+	  for the APP core the child image will be build for the NET core and
+	  vice versa.
+
+config NCS_CHILD_IMAGE_SAMPLE_EMPTY_APP_CORE
 	bool "Enable empty_app_core as a child image"
-	default n
 	depends on SOC_NRF5340_CPUNET
-	select PARTITION_MANAGER_ENABLED
+	imply AUTO_INCLUDE_CHILD_IMAGE
 	help
-	  Add the empty_app_core as a child image for the given sample.
+	  Add the empty_app_core as a child image on the application core.
 	  Used by samples that only require SOC_NRF5340_CPUNET.
 
+config NCS_CHILD_IMAGE_SAMPLE_MULTIPROTOCOL_RPMSG
+	bool "Multiprotocol RPMsg sample"
+	default y if BT_RPMSG_NRF53 && NRF_802154_SER_HOST
+	imply AUTO_INCLUDE_CHILD_IMAGE
+	help
+	  Add the multiprotocol_rpmsg sample as a child image on the network core.
+
+config NCS_CHILD_IMAGE_SAMPLE_HCI_RPMSG
+	bool "HCI RPMsg sample"
+	default y if BT_RPMSG_NRF53
+	imply AUTO_INCLUDE_CHILD_IMAGE
+	help
+	  Add the multiprotocol_rpmsg sample as a child image on the network core.
+
+config NCS_CHILD_IMAGE_SAMPLE_802154_RPMSG
+	bool "nRF IEEEE 802.15.4 Serialization RPMsg sample"
+	default y if NRF_802154_SER_HOST
+	imply AUTO_INCLUDE_CHILD_IMAGE
+	help
+	  Add the multiprotocol_rpmsg sample as a child image on the network core.
+
+config NCS_CHILD_IMAGE_SAMPLE_RPC_HOST
+	bool "nRF RPC Host sample"
+	default y if BT_RPC
+	imply AUTO_INCLUDE_CHILD_IMAGE
```

Comment:
I tried to use it, and it looks better from the user's point of view / menuconfig UI. I did have to do a teeny bit of condition duplication though, hope it's alright as I'd really like to be able to merge this if CI is green (been a few months already).
Unrelated: I clicked the `resolve conversation` button because I'm still not super accustomed to the github PR workflow and wasn't sure you'd get notified if I just posted the comment.